cdc/processor/etcdkey.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	"log"
+	"strings"
+
+	"github.com/pingcap/ticdc/cdc/model"
+	cerror "github.com/pingcap/ticdc/pkg/errors"
+)
+
+const (
+	etcdKeyBase = "/tidb/cdc"
+	ownerKey    = "/owner"
+	captureKey  = "/capture"
+
+	taskKey         = "/task"
+	taskWorkloadKey = taskKey + "/workload"
+	taskStatusKey   = taskKey + "/status"
+	taskPositionKey = taskKey + "/position"
+
+	changefeedInfoKey = "/changefeed/info"
+	jobKey            = "/job"
+)
+
+// CDCEtcdKeyType is the type of etcd key
+type CDCEtcdKeyType = int
+
+// the types of etcd key
+const (
+	CDCEtcdKeyTypeUnknown CDCEtcdKeyType = iota
+	CDCEtcdKeyTypeOnwer
+	CDCEtcdKeyTypeCapture
+	CDCEtcdKeyTypeChangefeedInfo
+	CDCEtcdKeyTypeChangeFeedStatus
+	CDCEtcdKeyTypeTaskPosition
+	CDCEtcdKeyTypeTaskStatus
+	CDCEtcdKeyTypeTaskWorkload
+)
+
+// CDCEtcdKey represents a etcd key which is defined by TiCDC
+type CDCEtcdKey struct {
```

Comment:
How will this struct be used, supposing capture wants to write a key `"/tidb/cdc/capture/6bbc01c8-0605-4f86-a0f9-b3119109b225"`, should it be constructed from a raw string or generate a `CDCEtcdKey` struct. Please add some regulations about it.